### PR TITLE
Graph traverse

### PR DESCRIPTION
**Pull request by jeremytedwards**

### CODE REVIEW

File: src/data_structures/simpleg.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        return end in self._dict[start]
+
+    def depth_first_traversal(self, start):
+        """
+        Perform a full depth-first traversal of the graph beginning at start.
+        Return the full visited path when traversal is complete.
+        """
+        visited = set()
+        stack = [start]
+        # let's do this without call recursion
```
Comment (nhuntwalker) [inline src/data_structures/simpleg.py:74]: Why not just use your Stack() object? I mean you went through all the trouble of creating one...

File: src/data_structures/simpleg.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        while stack:
+            node = stack.pop()
+            if node not in visited:
+                visited.add(node)
+                yield node
+                for neighbor in self.neighbors(node):
+                    stack.append(neighbor)
+
```
Comment (nhuntwalker) [inline src/data_structures/simpleg.py:82]: g.depth_first_traversal isn't currenly returning anything

File: src/data_structures/simpleg.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+        """
+        visited = set()
+        q = deque((start,))
+        while q:
+            node = q.pop()
+            if node not in visited:
+                visited.add(node)
+                yield(node)
+                for neighbor in self.neighbors(node):
+                    q.appendleft(neighbor)
```
Comment (nhuntwalker) [inline src/data_structures/simpleg.py:113]: Like your first DFS method this isn't returning anything.

Comment (mumbleskates) [inline src/data_structures/simpleg.py:113]: They're generators!

**Pull request merged**